backend/app/routes/captions.js

```diff
@@ -58,12 +58,21 @@ const voteCaption = {
             updateQuery = 'UPDATE CAPTION SET VOTE_COUNT=VOTE_COUNT-1 WHERE ID=?';
         }
 
-        return databaseUtil.sendQuery(updateQuery, [captionId])
-            .then(() => reply.response({ code: 1 }).code(200))
-            .catch((error) => {
-                console.log(error);
-                return reply.response({ code: 3 }).code(500);
-            });
+        // Check if the caption exists in db
+        const checkQuery = 'SELECT * FROM CAPTION WHERE ID=?';
+        return databaseUtil.sendQuery(checkQuery, [captionId]).then((result) => {
```

Comment:
You should refactor the code so that the promises are chained. See https://github.com/justin-cotarla/CaptionThis/pull/96/files